airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py

```diff
@@ -209,11 +282,10 @@ def upload_metadata_to_gcs(bucket_name: str, metadata_file_path: Path, validator
     Returns:
         Tuple[bool, str]: Whether the metadata file was uploaded and its blob id.
     """
-    if validator_opts.prerelease_tag:
-        metadata_file_path = create_prerelease_metadata_file(metadata_file_path, validator_opts)
 
-    metadata, error = validate_and_load(metadata_file_path, POST_UPLOAD_VALIDATORS, validator_opts)
+    metadata_file_path = _apply_modifications_to_metadata_file(metadata_file_path, validator_opts)
```

Comment:
I would find it more explicit and maintainable to decouple metadata mutation from metadata upload. Wdyt about create a separate step in publish which would perform this `apply_modifications_to_metadata_file`?

Reply:
Hmm I dont think I like the logic living in the pipeline.

Simply because right now we have metadata logic living already in two places
1. Metadata service (validation and upload)
2. Orchestrator (Parsing and generation)

Currently they are at least contained in the `metadata_service` folder

But if we add the git logic of
> for a given metadata file get the author of the last commit that modified

Then we end up with 3 locations to look for logic, 1 of which is now outside the area in the codebase